marimo/_runtime/runner/hooks_post_execution.py

```diff
@@ -79,6 +86,38 @@ def _broadcast_datasets(
         Datasets(tables=tables).broadcast()
 
 
+def _broadcast_duckdb_tables(
+    cell: CellImpl,
+    runner: cell_runner.Runner,
+    run_result: cell_runner.RunResult,
+) -> None:
+    del run_result
+    del runner
+    if not DependencyManager.has_duckdb():
+        return
+
+    try:
+        code = cell.code
+        visitor = SQLVisitor()
+        visitor.visit(ast.parse(code))
+        sqls = visitor.get_sqls()
```

Comment:
This should happen at cell creation time — can add a new attribute called `sqls` to `CellImpl`.